internal/interface/crawler/crawler.go

```diff
@@ -66,6 +109,108 @@ func (c *Crawler) getArticlesCount() (int, error) {
 	return 0, fmt.Errorf("unexpected query result")
 }
 
+func (c *Crawler) GetURLFromDB(ctx context.Context, URLChan chan<- string) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			url, err := c.GetUnvisitedURL()
+			if err == ErrEmptyQueue {
+				time.Sleep(time.Second)
+			} else if err != nil {
+				return err
+			} else {
+				totalURLsVisited.Inc()
```

Comment:
Concurent visits of the same URL will aggregate here as well.. Statistics will lie to us, as usual..

Reply:
As I said earlier, there's no way we visit the same URL twice.

Reply:
And there's no way this code could pass a review)) Unless Nikita R gives his blessing..